libcmis fails to build against boost-1.86.0

```
   repository:               https://github.com/tdf/libcmis.git
   at the commit:            52ba6c85831f561ddae9535c981bfb7fbb62e97b
...

checking for Boost headers version >= 1.36.0... yes
checking for Boost's header version... 1_86
checking for the toolset name used by Boost for x86_64-pc-linux-gnu-g++... configure: WARNING: could not figure out which toolset name to use for x86_64-pc-linux-gnu-g++

checking for boost/date_time/posix_time/posix_time.hpp... yes
checking for the Boost date_time library... yes
checking for boost/scoped_ptr.hpp... yes
checking for boost/shared_ptr.hpp... yes
checking for boost/algorithm/string.hpp... yes
checking for boost/uuid/uuid.hpp... yes
...

libtool: compile:  x86_64-pc-linux-gnu-g++ -DPACKAGE_NAME=\"libcmis\" -DPACKAGE_TARNAME=\"libcmis\" -DPACKAGE_VERSION=\"0.6.2\" "-DPACKAGE_STRING=\"libcmis 0.6.2\"" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libcmis\" -DVERSION=\"0.6.2\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_CXX11=1 -DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1 -DHAVE_BOOST=1 -DHAVE_BOOST_DATE_TIME_POSIX_TIME_POSIX_TIME_HPP=1 -DHAVE_BOOST_SCOPED_PTR_HPP=1 -DHAVE_BOOST_SHARED_PTR_HPP=1 -DHAVE_BOOST_ALGORITHM_STRING_HPP=1 -DHAVE_BOOST_UUID_UUID_HPP=1 -DHAVE_STDLIB_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDBOOL_H=1 -I. -DLIBCMIS_BUILD -DLIBCMIS_VISIBILITY -I../../inc -I../../src/libcmis -I/usr/include/libxml2 -fvisibility=hidden -march=native -mtune=native -O2 -pipe -Wall -pedantic -Weffc++ -Wshadow -Wendif-labels -Wextra -Wsign-promo -Woverloaded-virtual -Wnon-virtual-dtor -Wsign-promo -c ws-versioningservice.cxx  -fPIC -DPIC -o .libs/libcmis_la-ws-versioningservice.o
/bin/sh ../../libtool  --tag=CXX   --mode=compile x86_64-pc-linux-gnu-g++ -DPACKAGE_NAME=\"libcmis\" -DPACKAGE_TARNAME=\"libcmis\" -DPACKAGE_VERSION=\"0.6.2\" -DPACKAGE_STRING=\"libcmis\ 0.6.2\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libcmis\" -DVERSION=\"0.6.2\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_CXX11=1 -DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1 -DHAVE_BOOST=1 -DHAVE_BOOST_DATE_TIME_POSIX_TIME_POSIX_TIME_HPP=1 -DHAVE_BOOST_SCOPED_PTR_HPP=1 -DHAVE_BOOST_SHARED_PTR_HPP=1 -DHAVE_BOOST_ALGORITHM_STRING_HPP=1 -DHAVE_BOOST_UUID_UUID_HPP=1 -DHAVE_STDLIB_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDBOOL_H=1 -I.  -DLIBCMIS_BUILD -DLIBCMIS_VISIBILITY  -I../../inc -I../../src/libcmis -I/usr/include/libxml2   -fvisibility=hidden -march=native -mtune=native -O2 -pipe -Wall -pedantic -Weffc++ -Wshadow -Wendif-labels -Wextra -Wsign-promo -Woverloaded-virtual -Wnon-virtual-dtor -Wsign-promo -c -o libcmis_la-xml-utils.lo `test -f 'xml-utils.cxx' || echo './'`xml-utils.cxx
libtool: compile:  x86_64-pc-linux-gnu-g++ -DPACKAGE_NAME=\"libcmis\" -DPACKAGE_TARNAME=\"libcmis\" -DPACKAGE_VERSION=\"0.6.2\" "-DPACKAGE_STRING=\"libcmis 0.6.2\"" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libcmis\" -DVERSION=\"0.6.2\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_CXX11=1 -DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1 -DHAVE_BOOST=1 -DHAVE_BOOST_DATE_TIME_POSIX_TIME_POSIX_TIME_HPP=1 -DHAVE_BOOST_SCOPED_PTR_HPP=1 -DHAVE_BOOST_SHARED_PTR_HPP=1 -DHAVE_BOOST_ALGORITHM_STRING_HPP=1 -DHAVE_BOOST_UUID_UUID_HPP=1 -DHAVE_STDLIB_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDBOOL_H=1 -I. -DLIBCMIS_BUILD -DLIBCMIS_VISIBILITY -I../../inc -I../../src/libcmis -I/usr/include/libxml2 -fvisibility=hidden -march=native -mtune=native -O2 -pipe -Wall -pedantic -Weffc++ -Wshadow -Wendif-labels -Wextra -Wsign-promo -Woverloaded-virtual -Wnon-virtual-dtor -Wsign-promo -c xml-utils.cxx  -fPIC -DPIC -o .libs/libcmis_la-xml-utils.o
xml-utils.cxx: In function 'std::string libcmis::sha1(const std::string&)':
xml-utils.cxx:535:26: error: cannot convert 'unsigned int [5]' to 'unsigned char (&)[20]'
  535 |         sha1.get_digest( digest );
      |                          ^~~~~~
      |                          |
      |                          unsigned int [5]
In file included from xml-utils.cxx:40:
/usr/include/boost/uuid/detail/sha1.hpp:179:43: note:   initializing argument 1 of 'void boost::uuids::detail::sha1::get_digest(unsigned char (&)[20])'
  179 | inline void sha1::get_digest(digest_type& digest)
      |                              ~~~~~~~~~~~~~^~~~~~
make[2]: *** [Makefile:1207: libcmis_la-xml-utils.lo] Error 1
make[2]: Leaving directory '/var/tmp/portage/dev-cpp/libcmis-9999/work/libcmis-9999/src/libcmis'
make[1]: *** [Makefile:524: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/dev-cpp/libcmis-9999/work/libcmis-9999/src'
make: *** [Makefile:545: all-recursive] Error 1
 * ERROR: dev-cpp/libcmis-9999::gentoo failed (compile phase):
 *   emake failed
```

--- a/src/libcmis/xml-utils.cxx
+++ b/src/libcmis/xml-utils.cxx
@@ -531,16 +531,22 @@ namespace libcmis
         boost::uuids::detail::sha1 sha1;
         sha1.process_bytes( str.c_str(), str.size() );
 
-        unsigned int digest[5];
+        // on boost <  1.86.0, digest_type is typedef'd as unsigned int[5]
+        // on boost >= 1.86.0, digest_type is typedef'd as unsigned char[20]
+        boost::uuids::detail::sha1::digest_type digest;
         sha1.get_digest( digest );
 
+        // by using a pointer to unsigned char, we can read the
+        // object representation of either typedef.
+        const unsigned char* ptr = reinterpret_cast<const unsigned char*>( digest );
+
         stringstream out;
-        // Setup writing mode. Every number must produce eight
+        // Setup writing mode. Every number must produce two
         // hexadecimal digits, including possible leading 0s, or we get
         // less than 40 digits as result.
         out << hex << setfill('0') << right;
-        for ( int i = 0; i < 5; ++i )
-            out << setw(8) << digest[i];
+        for ( int i = 0; i < sizeof( digest ); ++ptr, ++i )
+            out << setw(2) << static_cast<int>( *ptr );
         return out.str();
     }
 

